### PR TITLE
Implement WriteStringToFile function

### PR DESCRIPTION
**Pull request by dimitry-**
For the cases when we want to write a string to file.

Test: run OrbitBaseTests

### CODE REVIEW

File: src/OrbitBase/include/OrbitBase/WriteStringToFile.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_WRITE_STRING_TO_FILE_H_
+#define ORBIT_BASE_WRITE_STRING_TO_FILE_H_
+
+#include <filesystem>
+#include <string>
```
Comment (beckerhe) [inline src/OrbitBase/include/OrbitBase/WriteStringToFile.h:9]: <string_view>

**Review by beckerhe — APPROVED** [review]
Otherwise LGTM

File: src/OrbitBase/include/OrbitBase/WriteStringToFile.h (hunk 1 above)
Comment (dimitry-) [reply src/OrbitBase/include/OrbitBase/WriteStringToFile.h:9]: done.

File: src/OrbitBase/WriteStringToFile.cpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+#if defined(__linux)
+#include <unistd.h>
+#elif defined(_WIN32)
+#include <io.h>
+// Windows does not have TEMP_FAILURE_RETRY - define a shortcut
+#define TEMP_FAILURE_RETRY(expression) (expression)
+#endif
+
+#include "OrbitBase//Logging.h"
```
Comment (dpallotti) [inline src/OrbitBase/WriteStringToFile.cpp:22]:
```suggestion
#include "OrbitBase/Logging.h"
```

File: src/OrbitBase/WriteStringToFile.cpp [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        if (fd != -1) close(fd);
+      });
+  if (fd == -1) {
+    return ErrorMessage(
+        absl::StrFormat("Unable to open file \"%s\": %s", file_name.string(), SafeStrerror(errno)));
+  }
+
+  ErrorMessageOr<void> result = WriteStringToFileImpl(fd, content);
+  if (!result) {
+    unlink(file_name.string().c_str());
```
Comment (dpallotti) [inline src/OrbitBase/WriteStringToFile.cpp:64]: I think we should use `remove`, `unlink` is not portable.

File: src/OrbitBase/WriteStringToFileTest.cpp [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  size_t content_size = 16;
+  const char* expected_content = "content\nnew line";
+  std::string_view content(full_content, content_size);
+
+  ErrorMessageOr<void> result = WriteStringToFile(temp_file_name, content);
+  ASSERT_TRUE(result) << result.error().message();
+  ErrorMessageOr<std::string> actual_content = ReadFileToString(temp_file_name);
+  ASSERT_TRUE(actual_content) << actual_content.error().message();
+  EXPECT_EQ(actual_content.value(), expected_content);
+  unlink(temp_file_name);  // TODO(http://b/180574275): see above
```
Comment (dpallotti) [inline src/OrbitBase/WriteStringToFileTest.cpp:29]: Same here, why not `remove`?

File: src/OrbitBase/WriteStringToFileTest.cpp [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+#include "OrbitBase/ReadFileToString.h"
+#include "OrbitBase/Result.h"
+#include "OrbitBase/WriteStringToFile.h"
+
+namespace orbit_base {
+TEST(WriteStringToFile, Smoke) {
+  // TODO(http://b/180574275): Replace this with temporary_file once it becomes available
+  const char* temp_file_name = std::tmpnam(nullptr);
+  const char* full_content = "content\nnew line(this text is not written)";
+  size_t content_size = 16;
```
Comment (dpallotti) [inline src/OrbitBase/WriteStringToFileTest.cpp:20]: Make it a constant?

**Review by dpallotti — APPROVED** [review]

File: src/OrbitBase/WriteStringToFile.cpp (hunk 2 above)
Comment (dimitry-) [reply src/OrbitBase/WriteStringToFile.cpp:22]: done.

File: src/OrbitBase/WriteStringToFile.cpp (hunk 3 above)
Comment (dimitry-) [reply src/OrbitBase/WriteStringToFile.cpp:64]: so the difference between remove and unlink is that remove can remove directories where unlink is specifically for files. It seems to be portable between linux and windows. what am I missing here?

File: src/OrbitBase/WriteStringToFileTest.cpp (hunk 5 above)
Comment (dimitry-) [reply src/OrbitBase/WriteStringToFileTest.cpp:20]: done.

File: src/OrbitBase/WriteStringToFile.cpp (hunk 3 above)
Comment (dpallotti) [reply src/OrbitBase/WriteStringToFile.cpp:64]: `unlink` is defined in `<unistd.h>`, that's enough of a sign for me. Yes, the same is provided on Windows in this case, but it's a deprecated alias: https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/unlink?view=msvc-160. 
So both work but `unlink` is the weird choice here.

Comment (dimitry-) [reply src/OrbitBase/WriteStringToFile.cpp:64]: same goes for open/read/write and remove for that matter. I think at this point we accept the fact that even thought microsoft is trying to deprecate posix there is enough users for this to be just a wish.

Comment (dpallotti) [reply src/OrbitBase/WriteStringToFile.cpp:64]: `remove` is defined in `<cstdio>`. For me the fact that we have to use posix functions for some things doesn't necessarily mean that we always have to do that even when there's an alternative.

Comment (dpallotti) [reply src/OrbitBase/WriteStringToFile.cpp:64]: This said you clearly seem to prefer `unlink` and this is really not important so in the end I'm fine with both.

Comment (dimitry-) [reply src/OrbitBase/WriteStringToFile.cpp:64]: tbh I do not see the point - they are both supported on windows and linux, on linux there is a clear difference between unlink and remove. but well, I do not have strong feeling here - will change to remove.

**Pull request merged**